Dockerfile.agent

```diff
@@ -4,15 +4,16 @@ MAINTAINER Kontena, Inc. <info@kontena.io>
 RUN apk update && apk --update add tzdata ruby ruby-irb ruby-bigdecimal \
     ruby-io-console ruby-json ca-certificates libssl1.0 openssl libstdc++
 
-ADD Gemfile /app/
-ADD Gemfile.lock /app/
+ADD common/kontena-common.gemspec /opt/kontena/common/
+ADD agent/Gemfile agent/Gemfile.lock /opt/kontena/agent/
 
 RUN apk --update add --virtual build-dependencies ruby-dev build-base openssl-dev && \
     gem install bundler --no-ri --no-rdoc && \
-    cd /app ; bundle install --without development test && \
+    cd /opt/kontena/agent/ ; bundle install --without development test && \
     apk del build-dependencies
 
-WORKDIR /app
-ADD . /app
+ADD common /opt/kontena/common
+ADD agent /opt/kontena/agent
 
-CMD ["/app/bin/kontena-agent"]
+WORKDIR /opt/kontena/agent
+CMD bundle exec /opt/kontena/agent/bin/kontena-agent
```

Comment:
Yes, we could, but I think it's better to have the `Dockerfile` be in the actual docker build context dir. That way it's apparent that you have to build it with `kontena $ docker build -t kontena/agent -f Dockerfile.agent .`. If it is just `agent/Dockerfile`, then the obvious thing of `kontena/agent $ docker build -t kontena/agent .` will fail.